data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1,5 +1,385 @@
+        description = {STR_VERSE_RESTRICT_{X}}
+    [/dummy]
+#enddef
+
+#define WEAPON_SPECIAL_ONCE_PER_TURN DISTINGUISHER
+# wmlxgettext: [attack]
+# wmlxgettext: [specials]
+    [disable]
+        id = once_per_turn_{DISTINGUISHER}
+        name = _ "once per turn"
+        description = _ "This attack can be used offensively only once per turn"
+        [filter_self]
+            formula = "self.wml_vars.{DISTINGUISHER}_used > 0"
+        [/filter_self]
+        active_on = offense
+    [/disable]
+[/specials]
+[/attack]
+    [event]
+        name = attack end
+        first_time_only = no
+        
+        [filter_attack]
+            special_id = once_per_turn_{DISTINGUISHER}
+        [/filter_attack]
+
+        [modify_unit]
+            [filter]
+                find_in = unit
+            [/filter]
+
+            [set_variable]
+                name = {DISTINGUISHER}_used
+                value = 1
+            [/set_variable]
+        [/modify_unit]
+    [/event]
+    [event]
+        name = turn end
+        first_time_only = no
+
+        [modify_unit]
+            [filter]
+                [has_attack]
+                    special_id = once_per_turn_{DISTINGUISHER}
+                [/has_attack]
+            [/filter]
+
+            [set_variable]
+                name = {DISTINGUISHER}_used
+                value = 0
+            [/set_variable]
+        [/modify_unit]
+    [/event]
+[+attack]
+[+specials]
+# wmlxgettext: [/specials]
+# wmlxgettext: [/attack]
+#enddef
+
+#define WEAPON_SPECIAL_NOVA
+# wmlxgettext: [attack]
+# wmlxgettext: [specials]
+    [dummy]
+        id = nova
+        name = _"nova"
+        description= _ "This weapon, when used offensively, deals damage to all units adjacent to the caster when it hits."
+        special_note=_ "This unit has wide-area attacks centered on the caster."
+        active_on = offense
+    [/dummy]
+[/specials]
+[/attack]
+    [event]
+        name = attacker_hits
+        id = nova
+        first_time_only = no
+        [filter_attack]
+            special_id = nova
+        [/filter_attack]
+        [harm_unit]
+            [filter]
+                [not]
+                    find_in = second_unit
+                [/not]
+                [filter_adjacent]
+                    find_in = unit
+                [/filter_adjacent]
+            [/filter]
+            amount = $weapon.damage
+            damage_type = $weapon.type
+            alignment = $unit.alignment
+            kill = yes
+            experience = yes
+            fire_event = yes
+        [/harm_unit]
+    [/event]
+[+attack]
+[+specials]
+# wmlxgettext: [/specials]
+# wmlxgettext: [/attack]
+#enddef
+
+#define WEAPON_SPECIAL_RAY
+# wmlxgettext: [attack]
+# wmlxgettext: [specials]
+    [dummy]
+        id = ray
+        name = _"ray"
+        description= _ "This weapon, when used offensively, also deals damage to a unit behind the target."
+        active_on = offense
+    [/dummy]
+[/specials]
+[/attack]
```

Comment:
We shouldn't be adding unbalanced macros for abilities in mainline - compare the `feeding` ability for example.

Reply:
A better example for comparison is probably the Merman brawler's stun, since it's also an attack-triggered ability.

Reply:
Just for me to understand this better. What is happening here is the tool reading ``[/specials]
[/attack]`` but not reading ``wmlxgettext: [attack] wmlxgettext: [specials]``, shich causes error for the tool and the way to prevent that is spliting the event and the ability / weapon special into two macros?

Reply:
The main reason isn't tooling, it's that we find the `[/specials][/attack]...events...[+attack][+specials]` style to be unpleasant to read. But also, it needs those comments for wmlxgettext.

In terms of how to do it, it might be time for @CelticMinstrel or I to write our preferred implementation, rather than to say that we can see a better way to do it.

@CelticMinstrel data/lua/stun.lua should have more comments, if it's going to be a reference document for how to implement abilities.

Reply:
> if it's going to be a reference document for how to implement abilities.

…why would it be a reference document? There's already a reference on the wiki for the game_events API, and there are several examples of abilities to look at including stun.lua.

Reply:
I mean, if it's the example that we refer people to then it should have more comments.

Reply:
@CelticMinstrel would it make sense for the schema to accept `[dummy]value=`?

@Hejnewar the commit that's auto-linked itself to this PR implements this:
```
#define UTBS_ABILITY_TAILWIND X
    [dummy]
        id = tailwind
        value = {X}
        name = _ "tailwind +"+{X}
        description = _ "At the start of the turn this unit increases movement points of surrounding units by +"+{X}
    [/dummy]
#enddef
```

Reply:
> would it make sense for the schema to accept `[dummy]value=`?

I don't think so; if you need a value for a "dummy" ability I think it makes more sense to take the approach in #7249 and use a custom tag which is then added to the schema as an ability with a value.

Reply:
@Hejnewar I've prototyped the implement-tailwind-as-a-single-event and event-via-resource in the couple of commits that have auto-linked to this PR, although the renaming from `[dummy]` to `[tailwind]` is still a todo. However I'm leaving it to you to work out what to do with the other abilities.